### ISSUE TITLE
🐛 noUnusedVariables false positive

### ISSUE THREAD

**Issue by probablyup** [biomejs/biome#281]
### Environment information

```block
LI:
  Version:                      1.1.2
  Color support:                true

Platform:
  CPU Architecture:             aarch64
  OS:                           macos

Environment:
  BIOME_LOG_DIR:                unset
  NO_COLOR:                     unset
  TERM:                         "xterm-256color"
  JS_RUNTIME_VERSION:           "v18.14.0"
  JS_RUNTIME_NAME:              "node"
  NODE_PACKAGE_MANAGER:         "yarn/3.2.1"

Biome Configuration:
  Status:                       Loaded successfully
  Formatter disabled:           true
  Linter disabled:              false
  Organize imports disabled:    true
  VCS disabled:                 true

Biome json:


{
  "$schema": "https://biomejs.dev/schemas/1.1.2/schema.json",
  "javascript": {
    "globals": [
      "expect",
      "describe",
      "it",
      "test",
      "google",
      "jest",
      "joint",
      "Chai",
      "cy",
      "Cypress",
      "beforeEach",
      "afterEach",
      "JSX",
      "CYPRESS_BUILD",
      "PLEASE_DO_NOT_USE_THIS_ANY_FOR_MIGRATION_ONLY",
      "IntersectionObserverInit",
      "Nullable",
      "VoidFunction",
      "ZendeskWidget"
    ]
  },
  "files": {
    "maxSize": 8092000,
    "ignore": [
      "[redacted]"
    ]
  },
  "organizeImports": {
    "enabled": false
  },
  "formatter": {
    "enabled": false
  },
  "linter": {
    "enabled": true,
    "rules": {
      "all": false,
      "correctness": {
        "all": true
      },
      "suspicious": {
        "all": true,
        "noArrayIndexKey": "off",
        "noExplicitAny": "off",
        "noPrototypeBuiltins": "off",
        "noRedeclare": "off",
        "noShadowRestrictedNames": "off"
      },
      "style": {
        "all": true,
        "noImplicitBoolean": "off",
        "noNamespace": "off",
        "noNegationElse": "off",
        "useBlockStatements": "off",
        "noInferrableTypes": "off",
        "noNonNullAssertion": "off",
        "useEnumInitializers": "off",
        "useShorthandArrayType": "off",
        "useSingleVarDeclarator": "off"
      }
    }
  }
}
```

### What happened?

For the following code, I'm getting a false positive warning for the `noUnusedVariables` rule:

```
// timerId: This variable is unused.
// biome[lint/correctness/noUnusedVariables]
const timerId = setInterval(() => {
  if (analyticsTracker.current.sent) {
    clearInterval(timerId);
    sendOpenedWidgetAnalytic(event);
  }
}, 500);
```

### Expected result

No false positive, the variable is in use within the internal closure.

### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct

**Comment by probablyup** [comment]
Actually I think this was fixed by #256 which hasn't shipped yet
